### PR TITLE
Restore all AuthoredHintViewRequest fields

### PR DESCRIPTION
**Pull request by Hamms**
A while back, (as a result of [this commit](https://github.com/code-dot-org/code-dot-org/commit/1e49ec6aa6bdae4755ab9397bb5acedcffa246a3#diff-ef2cede4ae8abc197a134a5077635d42R10)) the AuthoredHintViewRequestsController switched to a
whitelisting approach to input validation; unfortunately, it missed the
majority of the data fields in the model, causing all created AuthoredHintViewRequests to no longer be created with data as of Aug 22, 2016, 3:14 PM PDT. This PR adds them back.

### CODE REVIEW

File: dashboard/test/controllers/authored_hint_view_requests_controller_test.rb [hunk 1]
```diff
@@ -72,4 +72,43 @@ class AuthoredHintViewRequestsControllerTest < ActionController::TestCase
       end
     end
   end
+
+  test 'records all (optional) data fields' do
+    data = {
+      scriptId: @script.id,
+      levelId: @level.id,
+      hintId: "first",
```
Comment (ashercodeorg) [inline dashboard/test/controllers/authored_hint_view_requests_controller_test.rb]: super nit: Use single quote.

File: dashboard/test/controllers/authored_hint_view_requests_controller_test.rb [hunk 2]
```diff
@@ -72,4 +72,43 @@ class AuthoredHintViewRequestsControllerTest < ActionController::TestCase
...
+      finalTestResult: 13,
+      finalActivityId: 14,
+      finalLevelSourceId: 15
+    }
+
+    post(
+      :create,
+      params: {
+        hints: [data]
+      },
```
Comment (ashercodeorg) [inline dashboard/test/controllers/authored_hint_view_requests_controller_test.rb]: super nit: At a minimum, I'd suggest keeping all of `params` on one line (removing the line breaks). If it fits, the whole `post`.

File: dashboard/app/controllers/authored_hint_view_requests_controller.rb [hunk 3]
```diff
@@ -9,7 +9,31 @@ def create
...
+      :nextActivityId,
+      :nextLevelSourceId,
+      :finalTime,
+      :finalAttempt,
+      :finalTestResult,
+      :finalActivityId,
+      :finalLevelSourceId,
+    ]
+
+    hints = params.permit(hints: fields).require(:hints)
```
Comment (ashercodeorg) [inline dashboard/app/controllers/authored_hint_view_requests_controller.rb:35]: Note that my migration in #13232 will cause the activity references to no longer be necessary. For clarity, I'll add dropping those columns to the migration PR.

Comment (Hamms) [reply dashboard/app/controllers/authored_hint_view_requests_controller.rb:35]: excellent, thank you!

**Pull request merged**